### ISSUE TITLE
Error when I change the data format to 'NCHW' for InceptionResnetV2 algorithm

### ISSUE THREAD

**Issue by saranyagopal1** [tensorflow/models#2580]
### System information
- **What is the top-level directory of the model you are using**: slim
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes (to change data format to NCHW)
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 14.04
- **TensorFlow installed from (source or binary)**: from source with MKL config (latest code synced on 23rd October, 2017)
- **TensorFlow version (use command below)**: 1.4.0rc0
- **Bazel version (if compiling from source)**: bazel was not compiled from source
- **CUDA/cuDNN version**: NA
- **GPU model and memory**:NA
- **Exact command to reproduce**: 

### Describe the problem
When I change the data format to NCHW for optimizing the model to run on CPU, I hit a run-time error:
InvalidArgumentError (see above for traceback): ConcatOp : Dimensions of inputs should match: shape[0] = [32,35,35,96] vs. shape[1] = [32,35,35,64]

This is what I did to change the data_format:
1. I did this after getting the images variable in both training and eval script:
images = tf.transpose(images, [0, 3, 1, 2])
2. I added data_format='NCHW' in slim.arg_scope in inception_resnet_v2.py:
with slim.arg_scope([slim.conv2d, slim.max_pool2d, slim.avg_pool2d],
                        stride=1, padding='SAME', data_format='NCHW'):
3. In all layers of inception, I did concat with 1 instead of 3.
[0001-Change-data_format-to-NCHW-for-optimized-run-on-CPU.txt](https://github.com/tensorflow/models/files/1409534/0001-Change-data_format-to-NCHW-for-optimized-run-on-CPU.txt)
 
Then I hit the above error when I run the training script:
 python train_image_classifier.py --train_dir=train1 --dataset_dir=data --dataset_name=flowers --dataset_split_name=train --model_name=inception_resnet_v2 --use-cpu

### Source code / logs
Attached both patch and logs.
[error_inc_res_v2.txt](https://github.com/tensorflow/models/files/1409487/error_inc_res_v2.txt)
[0001-Change-data_format-to-NCHW-for-optimized-run-on-CPU.txt](https://github.com/tensorflow/models/files/1409535/0001-Change-data_format-to-NCHW-for-optimized-run-on-CPU.txt)

**Comment by skye** [comment]
@nathansilberman @sguada

**Comment by shantianyang** [comment]
hi, @saranyagopal1 , how about the "ConcatOp : Dimensions of inputs should match"?  I met the same problem, Is there some solution to fix it? thanks.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
